universal-login-relayer/lib/services/messages/PendingMessagesSQLStore.ts

```
@@ -0,0 +1,105 @@
+import Knex from 'knex';
+import {Wallet, Contract} from 'ethers';
+import WalletContract from '@universal-login/contracts/build/WalletMaster.json';
+import {getKeyFromHashAndSignature} from '../../utils/utils';
+import {InvalidExecution} from '../../utils/errors';
+import IPendingMessagesStore from './IPendingMessagesStore';
+import PendingMessage from './PendingMessage';
+
+export class PendingMessagesSQLStore implements IPendingMessagesStore {
+  constructor(public knex: Knex) {
+  }
+
+  async add(messageHash: string, pendingMessage: PendingMessage) {
+    return this.knex
+      .insert({
+        messageHash,
+        transactionHash: pendingMessage.transactionHash,
+        walletAddress: pendingMessage.walletAddress,
+        createdAt: this.knex.fn.now()
+      })
+      .into('messages');
+  }
+
+  async get(messageHash: string) {
+    const message = await this.knex('messages')
+      .where('messageHash', messageHash)
+      .first();
+    const signatureKeyPairs = await this.getCollectedSignatureKeyPairs(messageHash);
+    const pendingMessage: PendingMessage = message && {
+      transactionHash: message.transactionHash,
+      walletAddress: message.walletAddress,
+      collectedSignatureKeyPairs: signatureKeyPairs
+    };
+    return pendingMessage;
+  }
+
+  async isPresent(messageHash: string) {
+    const message = await this.knex('messages')
+      .first()
+      .where('messageHash', messageHash);
+    const signatureKeyPairs = await this.knex('signatureKeyPairs')
```

There should be always at least one -> throw UnexpectedError if 0